packages/plugin-ext/src/main/browser/plugin-icon-theme-service.ts

```diff
@@ -540,14 +546,26 @@ export class PluginIconThemeService implements LabelProviderContribution {
     register(contribution: IconThemeContribution, plugin: DeployedPlugin): Disposable {
         const pluginId = getPluginId(plugin.metadata.model);
         const packagePath = plugin.metadata.model.packagePath;
+        const packageUri = (() => {
+            if (isWindows) {
```

Comment:
that's not what I meant, we should construct URIs on backend. I meant to deprecate `PluginModel.packagePath` with `PluginModel.packageUri`. In this class we can remove `packagePath` and replace it with `packageUri`. I don't think it is going to break anyone.

The issue is that in browser case `isWindows` return for client OS, not backend OS. So if your client is linux, but server runs on windows It won't work.

Reply:
Oh. I see. Then I'll update it again to construct the Uri on the backend and pass it on.

Reply:
Should be done